test/e2e_node/util.go

```diff
@@ -454,10 +449,16 @@ func stopKubelet() func() {
 	stdout, err = exec.Command("sudo", "systemctl", "kill", kubeletServiceName).CombinedOutput()
 	framework.ExpectNoError(err, "Failed to stop kubelet with systemctl: %v, %s", err, string(stdout))
 
+	// wait until the kubelet health check fail
+	gomega.Eventually(ctx, func() bool {
+		return kubeletHealthCheck(kubeletHealthCheckURL)
+	}, f.Timeouts.PodStart, f.Timeouts.Poll).Should(gomega.BeFalseBecause("kubelet was expected to be stopped but it is still running"))
+
 	return func() {
 		// we should restart service, otherwise the transient service start will fail
 		stdout, err := exec.Command("sudo", "systemctl", "restart", kubeletServiceName).CombinedOutput()
 		framework.ExpectNoError(err, "Failed to restart kubelet with systemctl: %v, %v", err, stdout)
+		waitForKubeletToStart(ctx, f)
```

Comment:
is it a good practice to preserver the same `ctx` object vs. returning the function that will receive it's own context object?

Reply:
This function already uses `kubeletServiceName` the same way, so I decided to use `ctx` the same way.
This function looks a bit awkward to use to me, to be honest. 
Would it be more clear to return `kubeletServiceName` from it and then call `startKubelet` explicitly from the test cases, e.g.

```
// util.go
func startKubelet(ctx context.Context, kubeletServiceName string) {
	// reset the kubelet service start-limit-hit
	stdout, err := exec.CommandContext(ctx, "sudo", "systemctl", "reset-failed", kubeletServiceName).CombinedOutput()
	framework.ExpectNoError(err, "Failed to reset kubelet start-limit-hit with systemctl: %v, %s", err, string(stdout))

	stdout, err = exec.CommandContext(ctx, "sudo", "systemctl", "start", kubeletServiceName).CombinedOutput()
	framework.ExpectNoError(err, "Failed to start kubelet with systemctl: %v, %s", err, string(stdout))
}

// test case
serviceName := stopKubelet(ctx, f)
...
startKubelet(ctx,serviceName)
```

Reply:
service name reuse is fine. I just don't like `ctx` reuse. I think it is frown upon to keep this context this way.

Reply:
done. Thank you for the review!